text/048-packaging-coq-native.md

```diff
@@ -0,0 +1,180 @@
+- Title: Packaging coq with native_compute
+
+- Drivers: Erik Martin-Dorel (**@erikmd**), Pierre Roux (**@proux01**)
+
+- URL: <https://github.com/coq/ceps/pull/48>
+
+----
+
+# Motivation
+
+Currently, the use of `native_compute` in libraries involving many dependencies is [inconvenient](https://github.com/coq/coq/issues/12564#issuecomment-647546401) (so that nobody actually uses this feature). Indeed, all their dependencies need to be recompiled with the `-native-compiler yes` option passed to `coqc`. And this needs to be done in a specific way for each dependency, depending on the build system it uses (see e.g., [the doc of `coq_makefile`](https://coq.inria.fr/refman/practical-tools/utilities.html#precompiling-for-native-compute)), if possible at all.
+
+The overall aim of this proposal is to provide an `opam` meta-package (named e.g. `coq-native`) that users can install to automatically (re)compile everything with `native_compute` (no need for any manual `opam pin edit`).
```

Comment:
Can you call this a "virtual package" instead of a "meta package"? Unless you have examples of packages that were similar to `coq-native` and were called that way. In the Coq platform, we have used the "meta package" to designate the fact that it would be an umbrella package whose only interest would be its dependencies. In the case of `coq-native`, this is not exactly the same concept.

Reply:
Thanks @Zimmi48 

> Can you call this a "virtual package" instead of a "meta package"?

After a [quick search in opam-repository](https://github.com/ocaml/opam-repository/search?q=%22meta-package%22), I found essentially one so-called "meta package": [tezos-base](https://github.com/ocaml/opam-repository/blob/7ce73dd042f5ae19f6c0c0807cd661297258cb11/packages/tezos-base/tezos-base.7.4/opam#L23), which mostly matches the semantics you described for the Coq platform (i.e., a "shortcut" for installing its dependencies).

On the other hand, "virtual package" is a widely used term in opam-repository, with [many more occurrences](https://github.com/ocaml/opam-repository/search?q=%22virtual%20package%22) -- and semantics (including, but not limited to, that of "meta package" :) -- but all in all, I see no objection of doing the renaming you suggest.